### PR TITLE
Blog: Docker image maintenance

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

Blog: Contributor Training, Docker Image Maintenance.

With minor changes to a couple other blogs that I discovered while writing

Review at 
* https://20240730-blog-maintain-docke.ddev-com-front-end.pages.dev/blog/ddev-docker-image-maintenance/
* https://20240730-blog-maintain-docke.ddev-com-front-end.pages.dev/blog/ddev-local-automated-testing/ (not much effort needed here, I just did minor updates)

### CODE REVIEW

File: src/content/blog/ddev-docker-image-maintenance.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+````dockerfile
+RUN touch /var/tmp/this-ran.txt
+````
+
+When your project starts up it will actually build this into the image. You can use this technique for texting and experimentation.
+
+### Adding conditional layers in Golang code
+
+DDEV's golang code adds a few layers to the images at the first `ddev start` on a project. You can see all of these consolidated in a project in the `.ddev/.webimageBuild/Dockerfile` (which is not something you will ever change, it's a generated file). That's where things like the matching in-container Linux user and group are added, and a few other things. You can see how DDEV adds these layers in [WriteBuildDockerfile()](https://github.com/ddev/ddev/blob/c2aca52a18687e678086dd232573cf51914dba56/pkg/ddevapp/config.go#L1113). It's unusual to do things with images this way, but some problems do require logic during the Golang processes.
```
Comment (stasadev) [inline src/content/blog/ddev-docker-image-maintenance.md:90]:
```suggestion
DDEV's Golang code adds a few layers to the images at the first `ddev start` on a project. You can see all of these consolidated in a project in the `.ddev/.webimageBuild/Dockerfile` (which is not something you will ever change, it's a generated file). That's where things like the matching in-container Linux user and group are added, and a few other things. You can see how DDEV adds these layers in [WriteBuildDockerfile()](https://github.com/ddev/ddev/blob/c2aca52a18687e678086dd232573cf51914dba56/pkg/ddevapp/config.go#L1113). It's unusual to do things with images this way, but some problems do require logic during the Golang processes.
```

File: src/content/blog/ddev-docker-image-maintenance.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+[`bats`](https://github.com/bats-core/bats-core) is a `bash`-based test framework and it's used here because the tests have nothing to do with Golang, and for most people the barrier to entry for a bash test is lower. `bats` is also used by default in DDEV add-on tests, and is covered in the [Contributor Training on Add-Ons](advanced-add-on-contributor-training.md). PRs that make nontrivial changes to the Docker images should normally have test coverage at this level.
+
+## Testing images with the Golang-based tests
+
+There are hundreds of Golang-based tests as well, some of which do a good job exercising Docker image features. This is a fine way to do this work. 
+
+## Resources
+
+- [Release Management and Docker Images](https://ddev.readthedocs.io/en/stable/developers/release-management/)
+- [DDEV Docker Architecture](https://ddev.com/blog/ddev-docker-architecture) contributor training.
```
Comment (stasadev) [inline src/content/blog/ddev-docker-image-maintenance.md:112]:
```suggestion
- [DDEV Docker Architecture](ddev-docker-architecture.md) contributor training.
```

**Review by stasadev — APPROVED** [review]
Great, I have a few minor suggestions.

File: src/content/blog/ddev-docker-image-maintenance.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+<div class="video-container">
+<iframe width="560" height="315" src="https://www.youtube.com/embed/HcJOm0nBU0w?si=M19qxd8ZoRjnHR9N" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+</div>
+
+DDEV depends on a [few critical Docker images](https://github.com/ddev/ddev/tree/master/containers), including `ddev-webserver`, `ddev-dbserver`, `ddev-traefik-router`, and `ddev-ssh-agent`. Each of these is built for new releases, and if a PR changes them, then a new image tag is pushed for that PR (typically with the same tag as the PR branch name).
+
+## Building images locally for testing your changes
+
+As a prerequisite to building multi-architecture Docker images, do `docker buildx create —use`, which sets up `docker buildx` to do that.
```
[the remark this answers is not in the document]
Comment (rfay) [reply src/content/blog/ddev-docker-image-maintenance.md:26]: Wow. In my head it was `-use` but in my fingers it's `--use`. Thanks.

**Comment by rfay** [comment]
Thanks so much @stasadev . I really appreciate you looking at these things. You have a great eye even when you don't necessarily have opinions about the content. It makes a big difference at least having somebody with an eagle eye look at it like you do.

**Pull request merged**